app/core/Tracker2/SegmentTracker.js

```diff
@@ -103,49 +114,92 @@ export default class SegmentTracker extends BaseTracker {
   }
 
   async _initializeTracker () {
+    this.watchForDisableAllTrackingChanges(this.store)
+
     this.store.watch(
       (state, getters) => getters['me/isTeacher'],
-      this.onIsTeacherChanged.bind(this)
+      () => this.onIsTeacherChanged()
     )
 
     if (this.store.getters['me/isTeacher']) {
-      loadSegment()
+      this.onIsTeacherChanged(true)
       window.analytics.ready(this.onInitializeSuccess)
     } else {
       this.onInitializeSuccess()
     }
   }
 
-  async trackPageView (includeIntegrations = {}) {
+  async trackPageView (includeIntegrations = []) {
     await this.initializationComplete
 
-    if (!this.enabled) {
+    if (!this.enabled || this.disableAllTracking) {
       return
     }
 
+    const options = { ...DEFAULT_SEGMENT_OPTIONS }
+    this.addIntegrationsToSegmentOptions(options, includeIntegrations)
+
     const url = `/${Backbone.history.getFragment()}`
-    window.analytics.page(url, {}, {
-      integrations: {
-        All: false,
-        'Google Analytics': false
-      }
+    return new Promise((resolve) => {
+      window.analytics.page(undefined, url, {}, options, resolve)
```

Comment:
Nitpick: The url is placed in the `name` argument, and I think the url is also sent in the properties object. Is there a benefit to track it twice?
My info is from the [segment docs](https://segment.com/docs/connections/sources/catalog/libraries/website/javascript/#page).
> A dictionary of properties of the page. Note: url, title, referrer and path are collected automatically! Additionally this defaults to a canonical url, if available, and falls back to document.location.href.

Reply:
Good question and I'm not sure.  I'm included to keep it the same as we had in the legacy tracker though as that will keep segment reporting in the same way and thus any integrations or triggers we had based on this will not be impacted.  Thoughts?

Reply:
I think it might be more explicit to use `name` instead of `url`, but don't mind.